utils/io/fileutils/files.go

```diff
@@ -400,3 +405,62 @@ func CopyDir(fromPath, toPath string, includeDirs bool) error {
 	}
 	return err
 }
+
+// Returns the path to the directory in which itemToFind is located.
+// Traversing through directories from current work-dir to root.
+// traverseType determines whether looking for a file or dir.
+func GetFileOrDirPath(itemToFInd string, traverseItemType TraverseItemType) (string, error) {
```

Comment:
I suggest renaming this function to *FindUpstream*.

Reply:
Renamed.